prow/deck/jobs/jobs.go

```diff
@@ -124,13 +138,19 @@ func (c *filteringProwJobLister) ListProwJobs(selector string) ([]prowapi.ProwJo
 
 	var filtered []prowapi.ProwJob
 	for _, item := range prowJobList.Items {
-		shouldHide := item.Spec.Hidden || c.pjHasHiddenRefs(item)
-		if shouldHide && c.showHidden {
-			filtered = append(filtered, item)
-		} else if shouldHide == c.hiddenOnly {
-			// this is a hidden job, show it if we're asked
-			// to only show hidden jobs otherwise hide it
+		if len(c.tenantIDs) != 0 && c.TenantIDMatch(item) {
+			// Deck has tenantID and it matches Prowjob
 			filtered = append(filtered, item)
+		} else if len(c.tenantIDs) == 0 {
+			// Deck has no tenantID
+			shouldHide := item.Spec.Hidden || c.pjHasHiddenRefs(item)
+			if shouldHide && (c.showHidden || c.hiddenOnly) {
+				// If Hidden and we are showing Hidden we add it
+				filtered = append(filtered, item)
+			} else if !shouldHide && !c.hiddenOnly && (item.Spec.ProwJobDefault == nil || item.Spec.ProwJobDefault.TenantID == "") {
```

Comment:
Yes. Prowjobs with a given tenantID will only show up on Deck instances with the same ID. The only exception to this is regardless of tenantID if it is marked as hidden, it will appear on a Deck instance with show-hidden or hiddenOnly.